interpreter/cling/lib/Interpreter/Value.cpp

```diff
@@ -135,16 +135,54 @@ namespace {
 namespace cling {
 
   Value::Value(const Value& other):
-    m_Storage(other.m_Storage), m_StorageType(other.m_StorageType),
+    m_Storage(other.m_Storage), m_NeedsManagedAlloc(other.m_NeedsManagedAlloc),
+    m_TypeKind(other.m_TypeKind),
     m_Type(other.m_Type), m_Interpreter(other.m_Interpreter) {
     if (other.needsManagedAllocation())
       AllocatedValue::getFromPayload(m_Storage.m_Ptr)->Retain();
   }
 
+  static Value::TypeKind getCorrespondingTypeKind(clang::QualType QT,
+                                                  const clang::ASTContext &C) {
+    using namespace clang;
+    if (const auto *ET = dyn_cast<EnumType>(QT.getTypePtr()))
+      QT = ET->getDecl()->getIntegerType();
+
+    if (C.hasSameType(QT, C.VoidTy))
+      return Value::kVoid;
+
+    if (!QT->isBuiltinType() || QT->castAs<BuiltinType>()->isNullPtrType())
```

Comment:
That's why I like the notion of "storage type" better than "corresponding type": for non-builtin types, the storage is through `kPtrTy`, while it's not obvious how a pointer is "corresponding". Consider renaming this (back) to `determineStorageType`, also for `Value::TypeKind` => `Value::StorageType`?